lib/improver/tests/nbhood/use_nbhood/test_CollapseMaskedNeighbourhoodCoordinate.py

```diff
@@ -0,0 +1,445 @@
+
+
+import unittest
+
+import iris
+from iris.tests import IrisTest
+from iris.exceptions import CoordinateNotFoundError
+from iris.coords import DimCoord
+
+import numpy as np
+
+from improver.nbhood.use_nbhood import CollapseMaskedNeighbourhoodCoordinate
+from improver.tests.nbhood.nbhood.test_BaseNeighbourhoodProcessing import (
+    set_up_cube)
+from improver.tests.nbhood.use_nbhood. \
+    test_ApplyNeighbourhoodProcessingWithAMask \
+    import set_up_topographic_zone_cube
+
+
+class Test__repr__(IrisTest):
+
+    """Test the __repr__ method of CollapseMaskedNeighbourhoodCoordinate."""
+
+    def test_basic(self):
+        """Test that the __repr__ method returns the expected string."""
+        coord_masked = "topographic_zone"
+        weights = iris.cube.Cube(np.array([1.0]), long_name="weights")
+        result = str(CollapseMaskedNeighbourhoodCoordinate(
+            coord_masked, weights))
+        msg = ("<ApplyNeighbourhoodProcessingWithAMask:"
+               " coord_masked: topographic_zone,"
+               " weights: weights / (unknown)"
+               "                 (-- : 1)>")
+        self.assertEqual(result, msg)
+
+
+class Test_reweight_weights(IrisTest):
+
+    """Test the reweight_weights function."""
+
+    def setUp(self):
+        """Set up a cube."""
+        self.cube = set_up_cube(
+            zero_point_indices=((0, 0, 2, 2),), num_grid_points=5)
+        self.cube = iris.util.squeeze(self.cube)
+        weights_data = np.array([[[0.8, 0.7, 0., 0, 0],
+                                  [0.7, 0.3, 0, 0, 0],
+                                  [0.3, 0.1, 0, 0, 0],
+                                  [0, 0, 0, 0, 0],
+                                  [0, 0, 0, 0, 0]],
+                                 [[0.2, 0.3, 1, 1, 1],
+                                  [0.3, 0.7, 1, 1, 1],
+                                  [0.7, 0.9, 1, 1, 1],
+                                  [1, 1, 1, 0.9, 0.5],
+                                  [1, 1, 1, 0.6, 0.2]],
+                                 [[0, 0, 0, 0, 0],
+                                  [0, 0, 0, 0, 0],
+                                  [0, 0, 0, 0, 0],
+                                  [0, 0, 0, 0.1, 0.5],
+                                  [0, 0, 0, 0.4, 0.8]]])
+        topographic_zone_points = [50, 150, 250]
+        topographic_zone_bounds = [[0, 100], [100, 200], [200, 300]]
+
+        weights_cubes = iris.cube.CubeList([])
+        for data, point, bounds in zip(weights_data, topographic_zone_points,
+                                       topographic_zone_bounds):
+            weights_cubes.append(
+                set_up_topographic_zone_cube(
+                    data, point, bounds, num_grid_points=5))
+        self.weights_cube = weights_cubes.merge_cube()
+        self.plugin = CollapseMaskedNeighbourhoodCoordinate("topographic_zone",
+                                                            self.weights_cube)
+
```

Comment:
Please add a test_basic to prove that the right classes of object are returned / modified by this process.

Reply:
I think it would also be worth adding a unit test for an array of all NaNs. This should expect a ValueError from normalise_weights.

Reply:
Done